File: src/main/java/com/github/unchama/targetedeffect/MessageEffects.kt
package com.github.unchama.targetedeffect

import org.bukkit.command.CommandSender

fun String.asMessageEffect() =
    TargetedEffect { commandSender: CommandSender ->
      commandSender.sendMessage(this@asMessageEffect)
    }

fun List<String>.asMessageEffect() =
  TargetedEffect { commandSender: CommandSender ->
    commandSender.sendMessage(this@asMessageEffect.toTypedArray())
  }

File: src/main/scala/com/github/unchama/targetedeffect/MessageEffects.scala
package com.github.unchama.targetedeffect

import com.github.unchama.targetedeffect
import org.bukkit.command.CommandSender

object MessageEffects {

  implicit class StringMessageEffect(val string: String) {
    def asMessageEffect(): TargetedEffect[CommandSender] =
      targetedeffect.delay(_.sendMessage(string))
  }

  implicit class StringListMessageEffect(val stringList: List[String]) {
    def asMessageEffect(): TargetedEffect[CommandSender] =
      targetedeffect.delay(_.sendMessage(stringList.toArray))
  }

}
